number_parser/parser.py

```diff
@@ -47,29 +47,63 @@ def _check_validity(current_token, previous_token, lang_data):
         if previous_token not in lang_data.big_powers_of_ten and previous_token is not None:
             return False
 
-    elif current_token in lang_data.big_powers_of_ten and previous_token in lang_data.big_powers_of_ten:
-        if lang_data.big_powers_of_ten[current_token] > lang_data.big_powers_of_ten[previous_token]:
+    elif current_token in lang_data.big_powers_of_ten:
+        power_of_ten = lang_data.big_powers_of_ten[current_token]
+        if power_of_ten < current_grp_value:
             return False
-
+        if total_value != 0 and previous_power_of_10 is not None:
+            if power_of_ten >= previous_power_of_10:
+                return False
     return True
 
 
+def _check_large_multiplier(current_token, total_value, current_grp_value, lang_data):
+    combined_value = total_value + current_grp_value
+    if combined_value == 0:
+        return False
+    if current_token in lang_data.big_powers_of_ten:
+        large_value = lang_data.big_powers_of_ten[current_token]
+        if large_value > combined_value and large_value != 100:
+            return True
+    return False
+
+
 def _build_number(token_list, lang_data):
     """Incrementaly builds a number from the list of tokens."""
     total_value = 0
     current_grp_value = 0
     previous_token = None
+    previous_power_of_10 = None
     value_list = []
+    used_skip_tokens = []
 
     for token in token_list:
-        if token.isspace():
+        if token.isspace() or token == "":
+            continue
+        if token in lang_data.skip_tokens:
+            used_skip_tokens.append(token)
+            continue
+
+        is_large_multiplier = _check_large_multiplier(token, total_value, current_grp_value, lang_data)
+        if is_large_multiplier:
+            combined_value = total_value + current_grp_value
+            total_value = combined_value * lang_data.big_powers_of_ten[token]
+            previous_token = token
+            current_grp_value = 0
+            used_skip_tokens = []
+            previous_power_of_10 = lang_data.big_powers_of_ten[token]
             continue
-        valid = _check_validity(token, previous_token, lang_data)
+
+        valid = _check_validity(token, previous_token, previous_power_of_10, total_value, current_grp_value, lang_data)
         if not valid:
             total_value += current_grp_value
             value_list.append(str(total_value))
             total_value = 0
             current_grp_value = 0
+            for skip_token in used_skip_tokens:
+                value_list.append(skip_token)
+            used_skip_tokens = []
+            previous_power_of_10 = None
```

Comment:
I'm not 100% confident, but it's a `continue` missing here? If that's not the case, you can remove the `used_skip_tokens = []` line as it's already reset below.